### PR TITLE
Added base component

### PR DESCRIPTION
**Pull request by H34D**
maybe we can use a base component instead of implementing different getters over and over again

- darkMode
- currentUser
- currentNetwork

### CODE REVIEW

**Comment by Velua** [comment]
@pingustar Suggested adding ```isAuthenticated``` to this bad boy. Shall we add that too? @H34D

**Comment by H34D** [comment]
> @pingustar Suggested adding `isAuthenticated` to this bad boy. Shall we add that too? @H34D
 yes

**Comment by H34D** [comment]
> @pingustar Suggested adding `isAuthenticated` to this bad boy. Shall we add that too? @H34D

done

**Comment by H34D** [comment]
its quite big tho

File: src/components/BaseComponent.vue [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { vxm } from "@/store";
+
+@Component
+export default class BaseComponent extends Vue {
+  get darkMode(): boolean {
+    return vxm.general.darkMode;
+  }
+
+  get currentUser(): string {
+    return vxm.wallet.currentUser;
```
Comment (Velua) [inline src/components/BaseComponent.vue:12]: Shall we adopt the new system of not referencing ```vxm``` directly? May as well do this now.

**Review by Velua — CHANGES_REQUESTED** [review]

File: src/components/BaseComponent.vue (hunk 1 above)
Comment (H34D) [reply src/components/BaseComponent.vue:12]: sounds good, i will take a look at it

Comment (H34D) [reply src/components/BaseComponent.vue]: done

**Review by Velua — APPROVED** [review]

**Pull request merged**